asteroidsyncservice/watch.cpp

```diff
@@ -133,3 +137,68 @@ void Watch::sendNotify(unsigned int id, QString appName, QString icon, QString b
 {
     m_iface->call("SendNotify", id, appName, icon, body, summary);
 }
+
+bool Watch::screenshotServiceReady()
+{
+    return fetchProperty("StatusScreenshotService").toBool();
+}
+
+unsigned int Watch::screenshotProgress()
+{
+    return m_scrnProgress;
+}
+
+void Watch::screenshotTransferProgress(unsigned int progress)
+{
+    m_scrnProgress = progress;
+    emit screenshotProgressChanged();
+}
+
+void Watch::screenshotReceived(QByteArray data)
+{
+    QString filePath = getScreenshotDir(m_screenshotUrl.dir()) + QDir::separator() + m_screenshotName;
+    QSaveFile file(filePath);
+    file.open(QIODevice::WriteOnly);
+    file.write(data);
+    file.commit();
+    m_screenshotPath = filePath;
+    emit screenshotPathChanged();
+}
+
+QString Watch::screenshotPath() const
+{
+    return m_screenshotPath;
+}
+
+void Watch::setScreenshotUrl(const QString url)
+{
+    QFileInfo fileInfo(url);
+    if(!createDir(fileInfo.dir()))
+        qDebug() << "Unable to create directory";
+    m_screenshotUrl = fileInfo;
+}
+
+bool Watch::createDir(const QDir path)
+{
+    if(!path.exists()) {
+        return path.mkdir(path.path());
+    } else {
+        return true;
+    }
+}
+
+QString Watch::getScreenshotDir(const QDir dir)
+{
+    return dir.path();
```

Comment:
Also why is there a function just for that?

Reply:
removed